File: backend/src/main/java/com/williamdsw/semsys/domain/dto/MeetingScheduleDTO.java
package com.williamdsw.semsys.domain.dto;

import java.io.Serializable;
import java.time.LocalDateTime;
import javax.validation.constraints.NotEmpty;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.williamdsw.semsys.domain.MeetingSchedule;

public class MeetingScheduleDTO implements Serializable {
	// FIELDS

	private static final long serialVersionUID = 1L;
	private Integer id;

	@JsonFormat(pattern = "yyyy-MM-dd HH:mm")
	private LocalDateTime datetime;

	private EmployeeDTO employee;
	private StudentDTO student;

	@NotEmpty(message = "Meeting Status is required!")
	private String meetingStatus;

	// CONSTRUCTORS

	public MeetingScheduleDTO() {
	}

	public MeetingScheduleDTO(MeetingSchedule schedule) {
		super();
		this.id = schedule.getId();
		this.datetime = schedule.getDatetime();
		this.employee = new EmployeeDTO(schedule.getEmployee());
		this.student = new StudentDTO(schedule.getStudent());
		this.meetingStatus = (schedule.getStatus() != null ? schedule.getStatus().getDescription() : null);
	}

	// GETTERS / SETTERS

	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public LocalDateTime getDatetime() {
		return datetime;
	}

	public void setDatetime(LocalDateTime datetime) {
		this.datetime = datetime;
	}

	public EmployeeDTO getEmployee() {
		return employee;
	}

	public void setEmployee(EmployeeDTO employeeDto) {
		this.employee = employeeDto;
	}

	public StudentDTO getStudent() {
		return student;
	}

	public void setStudent(StudentDTO studentDto) {
		this.student = studentDto;
	}

	public String getMeetingStatus() {
		return meetingStatus;
	}

	public void setMeetingStatus(String meetingStatus) {
		this.meetingStatus = meetingStatus;
	}
}

File: frontend/src/app/models/domain/meeting-schedule.dto.ts
import { EmployeeDTO } from './employee.dto';
import { StudentDTO } from './student.dto';

export class MeetingScheduleDTO {

    // FIELDS

    private id: number;
    private datetime: Date;
    private employee: EmployeeDTO;
    private student: StudentDTO;
    private meetingStatus: string;

    // CONSTRUCTOR

    constructor() {}

    // GETTERS / SETTERS

    public getId(): number {
        return this.id;
    }

    public setId(id: number) {
        this.id = id;
    }

    public getDatetime(): Date {
        return this.datetime;
    }

    public setDatetime(datetime: Date) {
        this.datetime = datetime;
    }

    public getEmployee(): EmployeeDTO {
        return this.employee;
    }

    public setEmployee(employee: EmployeeDTO) {
        this.employee = employee;
    }

    public getStudent(): StudentDTO {
        return this.student;
    }

    public setStudent(student: StudentDTO) {
        this.student = student;
    }

    public getMeetingStatus(): string {
        return this.meetingStatus;
    }

    public setMeetingStatus(meetingStatus: string) {
        this.meetingStatus = meetingStatus;
    }
}
